src/web_socket.toit

```diff
@@ -0,0 +1,304 @@
+// Copyright (C) 2022 Toitware ApS. All rights reserved.
+// Use of this source code is governed by an MIT-style license that can be
+// found in the LICENSE file.
+
+import binary show BIG_ENDIAN
+import net.tcp
+import reader
+import writer
+
+OPCODE_CONTINUATION_ ::= 0
+OPCODE_TEXT_         ::= 1
+OPCODE_BINARY_       ::= 2
+OPCODE_CLOSE_        ::= 8
+OPCODE_PING_         ::= 9
+OPCODE_PONG_         ::= 10
+FIN_FLAG_            ::= 0x80
+EIGHT_BYTE_SIZE_     ::= 127
+TWO_BYTE_SIZE_       ::= 126
+MAX_ONE_BYTE_SIZE_   ::= 125
+
+/**
+A WebSocket connection.
+A bidirectional socket connection capable of sending binary or text messages
+  according to RFC 6455.
+Obtained from the $Client.web_socket method.
+Currently the server side setup is not implemented.
+Currently does not implement ping and pong packets.
```

Comment:
I feel like this doesn't really belong into the toitdoc, but it probably doesn't hurt. (We might forget to remove the lines when we add the functions, though).